sim/dex.js

```diff
@@ -474,33 +477,27 @@ class ModdedDex {
 			return name;
 		}
 
-		let cached = this.effectCache.get(name);
-		if (!cached) {
-			cached = this.getEffectInner(name);
-			this.effectCache.set(name, cached);
+		let effect = this.effectCache.get(name);
+		if (effect) {
+			// @ts-ignore
+			return effect;
 		}
-		return cached;
-	}
 
-	/**
-	* While this function can technically return any kind of effect at
-	* all, that's not a feature TypeScript needs to know about.
-	* @param {string} name
-	* @return {PureEffect}
-	*/
-	getEffectInner(name) {
 		if (name.startsWith('move:')) {
 			// @ts-ignore
```

Comment:
```suggestion
```

I think your refactors have made `@ts-ignore` unnecessary except for at the returns.

Reply:
I thought that as well, but:

`sim/dex.js:487:4 - error TS2322: Type 'Move' is not assignable to type 'Format | Item | Template | Ability | ActiveMove | PureEffect | undefined'.
  Type 'Move' is missing the following properties from type 'ActiveMove': typeMod, hit`

Reply:
Should we maybe just make the Effect type include `Move` in addition to `ActiveMove`?

Reply:
Oh, because it's not an ActiveMove. Yeah, that makes sense; it's fine as is.

Reply:
I would not add `Move` to `Effect` because we rely on `effect.effectType === "Move"` to mean it's an ActiveMove (and therefore mutable).